src/pynxtools/nexus/nexus.py

```diff
@@ -55,15 +65,26 @@ def get_nx_class_path(hdf_info):
     if hdf_node.name == "/":
         return ""
     if isinstance(hdf_node, h5py.Group):
-        return (
-            get_nx_class_path(get_hdf_info_parent(hdf_info))
-            + "/"
-            + (
-                hdf_node.attrs["NX_class"]
-                if "NX_class" in hdf_node.attrs.keys()
-                else hdf_node.name.split("/")[-1]
+        try:
+            return (
+                get_nx_class_path(get_hdf_info_parent(hdf_info))
+                + "/"
+                + (
+                    hdf_node.attrs["NX_class"].decode("UTF-8")
+                    if "NX_class" in hdf_node.attrs.keys()
+                    else hdf_node.name.split("/")[-1]
+                )
+            )
+        except AttributeError:
+            return (
+                get_nx_class_path(get_hdf_info_parent(hdf_info))
+                + "/"
+                + (
+                    hdf_node.attrs["NX_class"]
+                    if "NX_class" in hdf_node.attrs.keys()
+                    else hdf_node.name.split("/")[-1]
+                )
```

Comment:
Maybe possible to shorten like this?
```suggestion
        nx_class = hdf_node.attrs.get("NX_class", hdf_node.name.split("/")[-1])
        if isinstance(nx_class, bytes):
            nx_class = nx_class.decode("UTF-8")
        return get_nx_class_path(get_hdf_info_parent(hdf_info)) + "/" + nx_class
```